File: csq/opencv4.py
import cv2
import numpy as np

# 1. 读取图片 + 转为RGB格式（OpenCV默认BGR，先转RGB）
img = cv2.imread("/home/csq/code/scripts/MDWHC.JPG") 
rgb_img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)

# 分离R、G、B通道并保存
r, g, b = cv2.split(rgb_img)

cv2.imshow("Single R Channel", r)
cv2.imshow("Single G Channel", g)
cv2.imshow("Single B Channel", b)

cv2.waitKey(0)
cv2.destroyAllWindows()

File: csq/opencv4.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace cv;
using namespace std;

int main() {
    Mat img = imread("/home/csq/code/scripts/MDWHC.JPG");
    if (img.empty()) {
        cout << "无法读取图片！" << endl;
        return -1;
    }

    // 2. 拆分BGR通道（OpenCV默认BGR顺序）
    vector<Mat> bgr_channels;
    split(img, bgr_channels);
    Mat R_channel = bgr_channels[2]; // 提取R通道
    Mat G_channel = bgr_channels[1]; // 提取G通道
    Mat B_channel = bgr_channels[0]; // 提取B通道

    // 3. 显示并保存单通道结果
    imshow("TASK4_R通道", R_channel);
    imshow("TASK4_G通道", G_channel);
    imshow("TASK4_B通道", B_channel);
    
    imwrite("task4_R_channel.png", R_channel);
    imwrite("task4_G_channel.png", G_channel);
    imwrite("task4_B_channel.png", B_channel);

    waitKey(0);
    destroyAllWindows();
    return 0;
}
